tsconfig.json

```diff
@@ -9,7 +9,8 @@
         "noEmit": false,
         "moduleResolution": "Node",
         "strict": true,
-        "noImplicitAny": true
+        "noImplicitAny": true,
+        "strictPropertyInitialization": false
```

Comment:
Why did you add this?

Reply:
without it typescript forces me to initialise all properties in the c'tor.

Reply:
That sounds like A Good Thing though ;-)

Reply:
I mean, the typescript docs don't seem to advise against them. And not having to type the name again in the c'tor seemed like a plus to me.
But if you feel strongly about it I'll remove and change the classes.

Reply:
No strong feeling, let's try it out.